src/NzbDrone.Core/ImportLists/Discogs/DiscogsWantlistParser.cs

```diff
@@ -0,0 +1,140 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using Newtonsoft.Json;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Common.Http;
+using NzbDrone.Common.Serializer;
+using NzbDrone.Core.ImportLists.Exceptions;
+using NzbDrone.Core.Parser.Model;
+
+namespace NzbDrone.Core.ImportLists.Discogs;
+
+public class DiscogsWantlistParser : IParseImportListResponse
+{
+    private ImportListResponse _importListResponse;
+    private IHttpClient _httpClient;
+    private DiscogsWantlistSettings _settings;
+
+    public DiscogsWantlistParser()
+    {
+    }
+
+    public void SetContext(IHttpClient httpClient, DiscogsWantlistSettings settings)
+    {
+        _httpClient = httpClient;
+        _settings = settings;
+    }
+
+    public IList<ImportListItemInfo> ParseResponse(ImportListResponse importListResponse)
+    {
+        _importListResponse = importListResponse;
+
+        var items = new List<ImportListItemInfo>();
+
+        if (!PreProcess(_importListResponse))
+        {
+            return items;
+        }
+
+        var jsonResponse = Json.Deserialize<DiscogsWantlistResponse>(_importListResponse.Content);
+
+        if (jsonResponse?.Wants == null)
+        {
+            return items;
+        }
+
+        foreach (var want in jsonResponse.Wants)
+        {
+            if (want?.BasicInformation != null)
+            {
+                try
+                {
+                    if (_httpClient != null && _settings != null)
+                    {
+                        var releaseInfo = FetchReleaseDetails(want.BasicInformation.ResourceUrl);
+                        if (releaseInfo != null)
+                        {
+                            items.Add(releaseInfo);
+                        }
+                    }
+                }
+                catch
+                {
+                    // If we can't fetch release details, skip this item
+                    continue;
```

Comment:
This should log what has failed as people will ask "why isn't x in" and there is no clear understanding of it from this